src/hba.c

```diff
@@ -644,8 +658,12 @@ static bool parse_ident_line(struct Ident *ident, struct TokParser *tp, int line
 	return true;
 
 failed:
+	/* Not null ident_map is not expected here. */
+	Assert(ident_map == NULL);
```

Comment:
This assert should be removed.  The `failed` block may run when unexpected tokens are encountered in addition to mem allocation failures. 

See

https://github.com/pgbouncer/pgbouncer/pull/1101/files#diff-c0d4b6bb6b74133fcf98cf505edbb8acac315acaa1a885f863b168771acb37efR592:~:text=%26map_name))%20%7B-,if%20(!expect(tp%2C%20TOK_IDENT%2C%20%26map_name))%20%7B,%7D,-map_name_copy%20%3D%20strdup

Reply:
I am sorry for this question - where ident_map is modified in this point?

I see only ` if (!expect(tp, TOK_IDENT, &map_name))`

Reply:
Yes, you are right it is not modified.  But still Assert seems unnecessary to me.  I guess, it demonstrates that no  
`free(ident_map)` is needed. But it serves no other purpose.

Reply:
I can save Assert and remove free(ident_map).

free(ident_map) is not required here and more over - it is incorrected.